mpcs/src/basefold/query_phase.rs

```diff
@@ -157,239 +134,172 @@ pub fn batch_verifier_query_phase<E: ExtensionField, Spec: BasefoldSpec<E>>(
-                            width: 2,
-                            // width is 2, thus height divide by 2 via right shift
-                            height: n_d_i >> 1,
-                        }],
-                        idx,
-                        slice::from_ref(&leafs),
-                        proof,
-                    )
-                    .expect("verify failed");
-                let coeff = <Spec::EncodingScheme as EncodingScheme<E>>::verifier_folding_coeffs(
-                    vp,
-                    log2_strict_usize(n_d_i) - 1,
-                    idx,
+                    idx >>= 1;
+                    mmcs_ext
+                        .verify_batch(
+                            pi_comm,
+                            &[Dimensions {
+                                width: 2,
+                                // width is 2, thus height divide by 2 via right shift
+                                height: 1 << log2_height,
+                            }],
+                            idx,
+                            slice::from_ref(&leafs),
+                            proof,
+                        )
+                        .expect("verify failed");
+                    let coeff = S::verifier_folding_coeffs(vp, log2_height, idx);
+                    folded = codeword_fold_with_challenge(&[leafs[0], leafs[1]], *r, coeff, inv_2);
+                }
+                assert!(
+                    final_codeword.values[idx] == folded,
+                    "final_codeword.values[idx] value {:?} != folded {:?}",
+                    final_codeword.values[idx],
+                    folded
                 );
-                folded = codeword_fold_with_challenge(&[leafs[0], leafs[1]], *r, coeff, inv_2);
-                n_d_i >>= 1;
-            }
-            debug_assert!(folding_sorted_order_iter.next().is_none());
-            assert!(
-                final_codeword.values[idx] == folded,
-                "final_codeword.values[idx] value {:?} != folded {:?}",
-                final_codeword.values[idx],
-                folded
-            );
-        },
-    );
+            },
+        );
     exit_span!(check_queries_span);
 
     // 1. check initial claim match with first round sumcheck value
-    assert_eq!(
-        // we need to scale up with scalar for witin_num_vars < max_num_var
-        dot_product::<E, _, _>(
-            batch_coeffs.iter().copied(),
-            point_evals.iter().zip_eq(circuit_meta.iter()).flat_map(
-                |((_, evals), CircuitIndexMeta { witin_num_vars, .. })| {
-                    evals.iter().copied().map(move |eval| {
-                        eval * E::from_canonical_u64(1 << (max_num_var - witin_num_vars) as u64)
-                    })
-                }
-            )
-        ),
-        { sumcheck_messages[0].evaluations[0] + sumcheck_messages[0].evaluations[1] }
-    );
+    // we need to scale up with scalar for witin_num_vars < max_num_var
```

Comment:
nitpicks: will be nice to update comment `witin_num_vars` with new naming accordingly

Reply:
Addressed in https://github.com/scroll-tech/ceno/pull/974/commits/f30ef0bd351cd6573044a2088a3c1e39f2d9544c.